httpx/dispatch/proxy_http.py

```diff
@@ -0,0 +1,186 @@
+import enum
+
+from ..concurrency.base import ConcurrencyBackend
+from ..config import (
+    DEFAULT_POOL_LIMITS,
+    DEFAULT_TIMEOUT_CONFIG,
+    CertTypes,
+    PoolLimits,
+    SSLConfig,
+    TimeoutTypes,
+    VerifyTypes,
+)
+from ..exceptions import ProxyError
+from ..middleware import basic_auth_str
+from ..models import (
+    URL,
+    AsyncRequest,
+    AsyncResponse,
+    Headers,
+    HeaderTypes,
+    Origin,
+    URLTypes,
+)
+from .connection import HTTPConnection
+from .connection_pool import ConnectionPool
+
+
+class HTTPProxyMode(enum.Enum):
+    DEFAULT = "DEFAULT"
+    FORWARD_ONLY = "FORWARD_ONLY"
+    TUNNEL_ONLY = "TUNNEL_ONLY"
+
+
+class HTTPProxy(ConnectionPool):
+    """A proxy that sends requests to the recipient server
+    on behalf of the connecting client. Not recommended for HTTPS.
```

Comment:
Why is it not recommended for HTTPS? (I see we *do* handle the HTTPS case, so perhaps there's a security guideline I'm missing? If so, maybe we can be more explicit about the reason here.)

Reply:
Ah, that comment no longer applies as CONNECT is the preferred method for HTTPS proxying. Before I merged the two PRs into one this comment was there.